### ISSUE TITLE
Error in sequence_file_in

### ISSUE THREAD

**Issue by joshuak94** [seqan/seqan3#395]
```cpp
struct my_traits : public sequence_file_in_default_traits_dna
{
    using sequence_alphabet = dna4;                        // instead of dna5

    template <typename alph>
    using sequence_container = bitcompressed_vector<alph>; // must be defined as a template!
};

sequence_file_in<my_traits> fin{"/tmp/my.fasta"};

//...
```

Produces errors about sequence_container_concept, among other strange things.

**Comment by eseiler** [comment]
Minimal example:

```cpp
#include <seqan3/alphabet/nucleotide/dna4.hpp>
#include <seqan3/range/container/bitcompressed_vector.hpp>
#include <seqan3/range/container/concatenated_sequences.hpp>
using namespace seqan3;

int main()
{
    using t = concatenated_sequences<bitcompressed_vector<dna4>>;
    t val;
    t val2;
    val.insert(val.begin(), val2.begin(), val2.end());
}
```

Fails with (most basic concept that fails):

```
/Users/enricoseiler/git/seqan3/include/seqan3/std/iterator:68:9: note: within 'template<class t> concept const bool std::Readable<t> [with t = seqan3::detail::random_access_iterator<seqan3::concatenated_sequences<seqan3::bitcompressed_vector<seqan3::dna4> > >]'
 concept Readable = static_cast<bool>(::ranges::Readable<t>());
         ^~~~~~~~
/Users/enricoseiler/git/seqan3/include/seqan3/std/iterator:68:9: note: 'static_cast<bool>(ranges::v3::concepts::Readable<ranges::v3::concepts::Readable, seqan3::detail::random_access_iterator<seqan3::concatenated_sequences<seqan3::bitcompressed_vector<seqan3::dna4>, std::vector<long unsigned int, std::allocator<long unsigned int> > > > >())' evaluated to false
```

If you use:

```cpp
using sequence_alphabet                 = dna4;
template <typename _sequence_alphabet>
using sequence_container                = bitcompressed_vector<_sequence_alphabet>;
template <typename _sequence_container>
using sequence_container_container      = std::vector<_sequence_container>;

template <typename _id_container>
using id_container_container            = std::vector<_id_container>;

template <typename _quality_container>
using quality_container_container       = std::vector<_quality_container>;
```

It'll compile

**Comment by h-2** [comment]
I'll have a look at it today!

**Comment by h-2** [comment]
fixed via #446
